openapi-cli/src/main/java/io/ballerina/openapi/cmd/BallerinaCodeGenerator.java

```diff
@@ -154,10 +142,35 @@ public void generateClientAndService(String definitionPath, String serviceName,
         //Update type definition list
         List<TypeDefinitionNode> preGeneratedTypeDefNodes = new ArrayList<>(
                 clientGenerator.getBallerinaAuthConfigGenerator().getAuthRelatedTypeDefinitionNodes());
-        List<TypeDefinitionNode> typeInclusionRecords = serviceGenerator.getTypeInclusionRecords();
         List<TypeDefinitionNode> typeDefinitionNodeList = clientGenerator.getTypeDefinitionNodeList();
-        preGeneratedTypeDefNodes.addAll(typeInclusionRecords);
         preGeneratedTypeDefNodes.addAll(typeDefinitionNodeList);
+        String serviceContent = "";
+        if (complexPaths.isEmpty()) {
+            OASServiceMetadata oasServiceMetadata = new OASServiceMetadata.Builder()
+                    .withOpenAPI(openAPIDef)
+                    .withFilters(filter)
+                    .withNullable(nullable)
+                    .withGenerateServiceType(generateServiceType)
+                    .withGenerateWithoutDataBinding(generateWithoutDataBinding)
+                    .build();
+            BallerinaServiceGenerator serviceGenerator = new BallerinaServiceGenerator(oasServiceMetadata);
+            serviceContent = Formatter.format
+                    (serviceGenerator.generateSyntaxTree()).toSourceCode();
+            sourceFiles.add(new GenSrcFile(GenSrcFile.GenFileType.GEN_SRC, srcPackage, srcFile,
```

Comment:
In the definition of the `GenSrcFile` class it seems like the `srcPackage` parameter is not properly set in the constructor.
```java
    public GenSrcFile(GenFileType type, String pkgName, String fileName, String content) {
        this.type = type;
        this.fileName = fileName;
        this.content = content;
    }
```

Is this intended ? If so why are we passing `srcPackage` to the constructor ?

Reply:
yeah there are unused parameters in the several constructor due to some code and feature changes happened in time to time therefore we have current revamp within repo,  this will be addressed within this. https://github.com/ballerina-platform/ballerina-library/issues/6056